### PR TITLE
Add basic map implementation

### PR DESCRIPTION
**Pull request by AlbanTrincherini**
Add a baseline map implementation that will probably be useful later. The project doesn't build because it doesn't have the private google API key which I don't want to put on github.

### CODE REVIEW

**Comment by BoyeGuillaume** [comment]
In order to put the API key on github you must create a secret. Then modify the CI so that it read the API key from an environment value (during testing). You can check [this post](https://github.com/cirruslabs/cirrus-ci-docs/discussions/866) for how to deal with secrets using Cirus CI

**Comment by Maeeen** [comment]
LGTM! Maybe specify the author of the images and/or the licenses.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -11,6 +11,26 @@
...
+             TODO: Before you run your application, you need a Google Maps API key.
+
+             To get one, follow the directions here:
+
+                https://developers.google.com/maps/documentation/android-sdk/get-api-key
+
+             Once you have your API key (it starts with "AIza"), define a new property in your
+             project's local.properties file (e.g. MAPS_API_KEY=Aiza...), and replace the
+             "YOUR_API_KEY" string in this file with "${MAPS_API_KEY}".
+        -->
```
Comment (BoyeGuillaume) [inline app/src/main/AndroidManifest.xml:25]:
```suggestion
```

Remove TODO because it's done ?

File: build.gradle [hunk 2]
```diff
@@ -2,4 +2,5 @@
 plugins {
     id 'com.android.application' version '7.4.1' apply false
     id 'com.android.library' version '7.4.1' apply false
+    id 'com.google.android.libraries.mapsplatform.secrets-gradle-plugin' version '2.0.1' apply false
```
Comment (BoyeGuillaume) [inline build.gradle:5]:
```suggestion
```

I don't believe you used that anywhere. Indeed the `./local.properties` file is already generated by the `.cirrus.yml` CI. Indeed this plugin is another way to expose the API key using a compiled binary. Anyway API keys are still recoverable by decompiling an APK so well...

**Review by BoyeGuillaume — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/geohunt/app/MapsActivity.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityMapsBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
```
Comment (Nicoslz) [inline app/src/main/java/com/github/geohunt/app/MapsActivity.java:33]: I don't think that you should cast here the return value to (SupportMapFragment), as you could also obtain a null value as the result of this line of code. Instead I think that you should have a if statement to control this case

File: build.gradle (hunk 2 above)
Comment (Nicoslz) [reply build.gradle:5]: The same as above

**Pull request closed without merging**